### PR TITLE
Solution in arm64 assembly, including CI

### PR DESCRIPTION
**Pull request by rbergen**
## Description

This is an implementation in arm64 assembly. Notes:

- The PR also includes a few optimizations and fixes in the amd64 assembly solution I submitted earlier. This makes that both solutions are algorithmically the same.
- I included a new version of CI.yml, to make it select and use arm64 as the build platform for specific solutions (this being the furst and currently only). 

## Contributing requirements

* [x] I read the contribution guidelines in CONTRIBUTING.md.
* [x] I placed my solution in the correct category folder.
* [x] I added a README.md or Dockerfile.
* [x] I selected `drag-race` as the target branch.
* [x] All code herein is licensed compatible with BSD-3.

### CODE REVIEW

**Comment by rbergen** [comment]
@marghidanu: I can also commit a CI.yml that checks for an "arch-arm64" flag file in the solution directory and selects the arm64 build if the file exists. If you'd like to take that approach, then I'll update the PR accordingly.

**Comment by rbergen** [comment]
The commit builds arm64 and/or amd64 based on respective flag files, defaulting to amd64 if no flag files are present.

**Comment by rbergen** [comment]
By the way, some unrequested information on why I trigger the arm64 and amd64 builds individually: I actually created a CI version that runs one, multi-arch build when both archs are requested (it's [here](https://github.com/rbergen/Primes/blob/arm64flagfile/.github/workflows/CI.yml) for your reference). I then ran into [this issue in buildx](https://github.com/docker/buildx/issues/59), however.

TL;DR: if you build a multi-arch image, you **must** --push it. Until that gets fixed (_if_ it gets fixed), the recommended approach is to build every arch individually for all purposes except production runs.

**Comment by marghidanu** [comment]
Looks good! Let's merge this
